docs/Advanced-Usage/Debugging-and-Profiling-on-FPGA/Dromajo.rst

```diff
@@ -0,0 +1,140 @@
+.. _dromajo:
+
+Dromajo Co-simulation with BOOM designs
+==================================================
+
+Instead of using TracerV to provide a cycle-by-cycle trace of a target
+CPU's architectural state, you can use the `Dromajo co-simulator <https://github.com/chipsalliance/dromajo>`_ to verify
+the functionality of a BOOM design.
+
+.. note:: This work is highly experimental. We hope to integrate this into FireSim in a cleaner fashion at a later point.
+
+.. note:: This work currently only works for single core BOOM designs.
+
+.. _dromajo-bridge:
+
+Building a Design with Dromajo
+-------------------------------
+
+In all FireChip designs, TracerV is included by default.
+However, Dromajo currently doesn't work with TracerV.
```

Comment:
1. I didn't try it.
2. Dromajo only works for single-core and it by default already dumps a trace so IMO TracerV is unneeded.

Reply:
It should work. Set Dromajo and TracerV's BridgeBinders to be ComposeIOBinder, instead of OverrideIOBinder, and the BridgeBinders should be able to be used together. This works because there is no host-to-target communication over the TracerV interface.